Lib/typing.py

```diff
@@ -227,16 +230,59 @@ def _collect_type_vars(types_, typevar_types=None):
     return tuple(tvars)
 
 
-def _check_generic(cls, parameters, elen):
-    """Check correct count for parameters of a generic cls (internal helper).
-    This gives a nice error message in case of count mismatch.
+def _check_type_parameter_count(cls, type_params):
```

Comment:
Can we leave out the runtime arity checking for `TypeVarTuple`? I'm in favor of keeping the runtime stuff minimal and letting the type checkers handle cases as needed. Otherwise, we're bound to miss some case and cause problems (or make it harder to extend this in the future).

Did you have a use case in mind for the runtime errors?

Reply:
> Did you have a use case in mind for the runtime errors?

Nah, nothing in particular, apart from the obvious thing of alerting the user if they make a mistake and don't have a type checker running for whatever reason.

It feels a bit weird to do arity checks for regular `TypeVar`s but not `TypeVarTuple`s, but _shrug_, I don't feel strongly about it, so have backed out the arity checks.